clvm/run_program.py

```diff
@@ -57,16 +64,35 @@ def traverse_path(sexp: SExp, env: SExp) -> Tuple[int, SExp]:
         cost = PATH_LOOKUP_COST_PER_LEG
         if sexp.nullp():
             return cost, sexp.null()
-        node_index = int_from_bytes(sexp.atom)
-        while node_index > 1:
+
+        b = sexp.atom
+
+        end_byte_cursor = 0
+        while end_byte_cursor < len(b) and b[end_byte_cursor] == 0:
+            end_byte_cursor += 1
+
+        cost += end_byte_cursor * PATH_LOOKUP_COST_PER_ZERO_BYTE
+        if end_byte_cursor == len(b):
+            return cost, sexp.null()
+
+        # create a bitmask for the most significant *set* bit
+        # in the last non-zero byte
```

Comment:
When you say "last non-zero byte", it might be more clear to say "least significant non-zero byte"

Reply:
yeah, I was thinking about this when I was reviewing this code. When I say "last" here, I mean the from the right, in a big-endian kind-of-way. It's the *first* non-zero byte in the buffer, because we're skipping leading zeroes. I made this mistake in my first revision, thinking about the number as little-endian